### ISSUE TITLE
Invalid "Expect" header

### ISSUE THREAD

**Issue by fyfey** [php-http/curl-client#18]
I would like to add the ability to handle a 100-continue status from the server as I just hit a bug where I would get the following response:

```http
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Date: Tue, 08 Mar 2016 10:34:43 GMT
Server: Apache/2.2.22 (Debian)
X-Powered-By: PHP/5.4.45-0+deb7u2
etc
```
It gets the initial header line and then throws an exception parsing `HTTP/1.1 200 OK` as a header due to the missing colon.

I fixed it by setting an empty "Expect" header. Somewhere an "Expect 100-continue" was being added to my request (I wasn't explicitly setting this).

Could we check the initial status line for this status, and then discard the first 2 lines and resume parsing the actual response?

**Comment by mekras** [comment]
The problem is that cURL sends the header "Expect: 100-continue" by default. But curl-client does not support the "Expect-Continue" mechanism. I'll suppress "Expect" header to resolve this bug. And if you need support for "Expect-Continue", please, create a separate issue for it.

**Comment by mekras** [comment]
Please, try this [Client.php](https://github.com/php-http/curl-client/blob/21885e9e1b8911ac07bd49e05fe3497966c37938/src/Client.php)

**Comment by fyfey** [comment]
Thanks mekras, I've just deployed the new Client and will let you know how testing goes

**Comment by fyfey** [comment]
This has indeed resolved the problems we were having. Thanks for the quick response!

**Comment by mekras** [comment]
Fixed in version [1.3](https://github.com/php-http/curl-client/releases/tag/v1.3.0).
